src/jobs/jobs.controller.ts

```diff
@@ -386,17 +492,6 @@ export class JobsController {
         );
       }
       return true;
-    } else if (jobParams.auth == JobsAuth.Authenticated) {
-      // verify that the user is authenticated
-      if (user === null) {
-        throw new HttpException(
-          {
-            status: HttpStatus.UNAUTHORIZED,
-          },
-          HttpStatus.UNAUTHORIZED,
-        );
-      }
-      return true;
     }
 
```

Comment:
If `auth` isn't an enum value then shouldn't it be interpreted as a group name? I forget what we decided on that, but it may need to be changed following this PR.

Reply:
I am not quite sure I understand this. So there are two other versions GROUP and USER in the documentation that a facility can specify in the jobs configuration. In the current version of the code, there is this a line in `instanceAuthorizationJobCreate` function: `ability.can(AuthOp.JobCreateConfiguration, jobInstance)` which is responsible for that use case. In casl factory it will check if the user that wants to create a job is in `jobCreateInstanceAuthorizationValues`. This one is defined as `[#all, #authenticated, username (of user trying to do a job) and his/her groups]`. So it kind of does what Spencer asks in casl.